File: IPRS.Server/Models/Department.cs
using System.ComponentModel.DataAnnotations;

namespace IPRS.Server.Models;

public class Department
{
    [Key] public int Id { get; set; }
    [Required, MaxLength(100)] public string Name { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Guid? ManagerId { get; set; }
    
    // Relationships
    public User? Manager { get; set; }
    public ICollection<User> Employees { get; set; } = new List<User>();

    public ICollection<PurchaseRequest> PurchaseRequests { get; set; } = new List<PurchaseRequest>();
}

File: iprs.client/src/schemas/department.ts
import { z } from 'zod';

export const departmentResponseSchema = z.object({
  id: z.number().int(),
  name: z.string(),
  createdAt: z.iso.datetime(),
  managerId: z.guid().nullable(),
});

export const createDepartmentSchema = z.object({
  name: z.string().min(1, 'Department name is required'),
  managerId: z.guid().nullable().optional(),
});

export const updateDepartmentSchema = z.object({
  name: z
    .string()
    .min(1, 'Department name cannot be empty')
    .nullable()
    .optional(),
  managerId: z.guid().nullable().optional(),
  removeManager: z.boolean().default(false),
});

